src/Chimplet/RSS.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Locomotive\Chimplet;
+
+use Locomotive\WordPress\Facade as WP;
+
+/**
+ * File: Chimplet RSS campaign management
+ *
+ * @package Locomotive\Chimplet
+ */
+
+/**
+ * Class: Chimplet Settings Page
+ *
+ * @version 2015-02-13
+ * @since   0.0.0 (2015-02-07)
+ */
+
+class RSS extends Base {
+
+	/**
+	 * @var WP  $wp  WordPress Facade
+	 */
+
+	public $wp;
+
+
+	/**
+	 * Constructor
+	 *
+	 * Prepares actions and filters for commodity methods.
+	 *
+	 * @access public
+	 * @param  WP  $wp  The WordPress Facade
+	 */
+
+	public function __construct( WP $wp = null )
+	{
+		$this->wp = ( $wp instanceof WP ? $wp : new WP );
+
+		$this->wp->add_action( 'init',              [ $this, 'init' ], 10 );
+		$this->wp->add_action( 'pre_get_posts',     [ $this, 'pre_get_posts' ], 10 );
+
+		if ( $this->get_option( 'mailchimp.campaigns.schedule.frequency' ) ) {
+			$this->wp->add_action( 'template_redirect', [ $this, 'generate_rss_feed' ], 1 );
+		}
+	}
+
+	/**
+	 * Register a rewrite endpoint for the RSS feed
+	 */
+
+	public function init() {
+		add_rewrite_tag( '%chimplet_schedule%', '(monthly|weekly|daily)' );
+		add_rewrite_rule( 'chimplet/(monthly|weekly|daily)/?$', 'index.php?chimplet_schedule=$matches[1]', 'top' );
+	}
+
+	/**
+	 * Filter returned posts for the RSS campaign
+	 *
+	 * @param $query
+	 */
+	public function pre_get_posts( $query ) {
+		$schedule = $query->get( 'chimplet_schedule' );
+
+		if ( empty( $schedule ) ) {
+			return;
+		}
+
+		$tax = is_array( $_GET['tax'] ) ? $_GET['tax'] : ''; //input var ok
+
+		if ( empty( $tax ) ) {
+			return;
+		}
+
+		$allowed_tax = $this->get_option( 'mailchimp.terms' );
+		$tax_query   = [];
+
+		foreach ( $tax as $tax_name => $tax_ids ) {
+
+			$tax_ids = explode( ',', $tax_ids );
+
+			foreach ( $tax_ids as $id ) {
+
+				if ( ! in_array( $id, $allowed_tax[ $tax_name ] ) ) {
+					// Something is not right... bail
+					return;
+				}
+			}
+
+			$tax_query[] = [
+				'taxonomy' => $tax_name,
+				'field'    => 'term_id',
+				'terms'    => $tax_ids,
+			];
+		}
+
+		// Build date_query
+		switch ( $schedule ) {
+			case 'monthly':
+				$date_query = [
+					[
```

Comment:
@jonathanbardo The date_query might not be necessary. This just needs to be a normal functioning RSS Feed (like any of the default ones), just limited to the users interests. MailChimp will intelligently handle if the posts in the feed fit the campaign's schedule.

Based on the current options value for `posts_per_rss` (e.g., 10) each automated campaign will include no more than 10 posts.